pkg/sql/catalog/schematelemetry/schema_telemetry_job.go

```diff
@@ -66,9 +100,62 @@ func (t schemaTelemetryResumer) Resume(ctx context.Context, execCtx interface{})
 		sql.LogExternally,
 		events...,
 	)
+
 	return nil
 }
 
+func processInvalidObjects(
+	ctx context.Context,
+	cfg *sql.ExecutorConfig,
+	asOf hlc.Timestamp,
+	metrics *Metrics,
+	maxRecords int,
+) error {
+	// TODO(before merge): is this the right txn type? Does this need to be run within a txn?
+	return sql.DescsTxn(ctx, cfg, func(ctx context.Context, txn isql.Txn, col *descs.Collection) (retErr error) {
+		err := txn.KV().SetFixedTimestamp(ctx, asOf)
+		if err != nil {
+			return err
+		}
+
+		rows, err := txn.QueryIteratorEx(ctx, "sql-telemetry-invalid-objects", txn.KV(), sessiondata.NodeUserSessionDataOverride, `SELECT id, error FROM "".crdb_internal.invalid_objects LIMIT `+strconv.Itoa(maxRecords))
+		if err != nil {
+			return err
+		}
+
+		defer func(it isql.Rows) {
+			retErr = errors.CombineErrors(retErr, it.Close())
+		}(rows)
+
+		var ok bool
+		count := int64(0)
+		for ok, err = rows.Next(ctx); ok; ok, err = rows.Next(ctx) {
```

Comment:
I feel like this for loop might be more readable like this:
```suggestion
		for {
			ok, err = rows.Next(ctx)
			if !ok {
				break
			}
```

Also, should the `err != nil` check happen before checking `ok`?